src/vs/base/browser/ui/iconLabel/iconLabelHover.ts

```diff
@@ -127,18 +135,28 @@ export function setupCustomHover(hoverDelegate: IHoverDelegate, htmlElement: HTM
 		});
 	};
 	const mouseOverDomEmitter = dom.addDisposableListener(htmlElement, dom.EventType.MOUSE_OVER, () => showHoverDelayed(hoverDelegate.delay), true);
-	return {
+	const hover: ICustomHover = {
 		show: () => {
 			showHoverDelayed(0); // show hover immediately
 		},
 		hide: () => {
 			hideHover(true, true);
 		},
+		update: newTooltip => {
+			markdownTooltip = newTooltip;
+			tooltip = getTooltipForCustom(newTooltip);
+
+			if (hoverWidget && !hoverWidget.isDisposed) {
+				hover.hide();
```

Comment:
I did not find a way to update an existing opened hover so this will result in a slight flicker, but on the other hand signals the user that the hover has changed.

Reply:
Makes sense to use hide/show.
`hover.show` calls `showHoverDelayed`. That seems wrong as it is used to set up a new hover and won't do anything if a hover is being prepared (or up) (hoverPreparation is set). 
We know the widget is up and all listeners to manage it are installed so I think that all that's needed is to dispose the existing widget and create a new one

-> 
```
const tooltip = getTooltipForCustom(markdownTooltip);
const resolvedTooltip = (await tooltip(tokenSource.token)) ?? (!isString(markdownTooltip) && !(markdownTooltip instanceof HTMLElement) ? markdownTooltip?.markdownNotSupportedFallback : undefined);

hoverWidget?.dispose();
hoverWidget = undefined;
if (resolvedTooltip) {
		if (resolvedTooltip && hoverPreparation) {
			const hoverOptions: IHoverDelegateOptions = {
				content: resolvedTooltip,
				target,
				showPointer: hoverDelegate.placement === 'element',
				hoverPosition: HoverPosition.BELOW,
				skipFadeInAnimation: true
			};

			hoverWidget = hoverDelegate.showHover(hoverOptions);
		}
}
```

Reply:
@aeschli I must admit, I am not so familiar with the logic of that code, can you maybe add that piece in a way that we reuse the logic and not duplicate it?

Btw I just resolved the merge conflict.